reco_utils/recommender/tfidf/tfidf_utils.py

```diff
@@ -0,0 +1,400 @@
+        """
+        try:
+            self.stop_words = self.tf.get_stop_words()
+        except:
+            self.stop_words = 'Run .tokenize_text() and .fit_tfidf() first'
+        return self.stop_words
+    
+    def __create_full_recommendation_dictionary(self, df_clean):
+        """ Create the full recommendation dictionary containing all recommendations for all items.
+        
+        Args:
+            df_clean (pd.DataFrame): Dataframe with cleaned text.
+        """
+
+        # Similarity measure
+        cosine_sim = linear_kernel(self.tfidf_matrix, self.tfidf_matrix)
+
+        results = {}
+        for idx, row in df_clean.iterrows():
+            similar_indices = cosine_sim[idx].argsort()[:-(len(df_clean)+1):-1]
+            similar_items = [(cosine_sim[idx][i], df_clean[self.id_col][i], df_clean[self.title_col][i]) for i in similar_indices]
+            results[row[self.id_col]] = similar_items[1:]
+        
+        # Save to class
+        self.recommendations = results
+    
+    def __organize_results_as_tabular(self, df_clean):
+        """ Restructures results dictionary into a table containing only the top k recommendations per item.
+        
+        Args:
+            df_clean (pd.DataFrame): Dataframe with cleaned text.
+        """
+        
+        # Initialize new dataframe to hold recommendation output
+        item_id = list()
+        title = list()
+
+        rec_rank = list()
+        rec_score = list()
+
+        rec_item_id = list()
+        rec_title = list()
+
+        for idx in range(0, len(self.recommendations)):
+            # Information about the item we are basing recommendations off of
+            rec_based_on = list(self.recommendations.keys())[idx]
+            tmp_item_id = str(df_clean.loc[df_clean[self.id_col] == rec_based_on][self.id_col].values[0])
+            tmp_title = str(df_clean.loc[df_clean[self.id_col] == rec_based_on][self.title_col].values[0])
+
+            # Iterate through top k recommendations
+            for i in range(0, self.k):
+                # Save to lists
+                item_id.append(tmp_item_id)
+                title.append(tmp_title)
+                rec_rank.append(str(i+1))
+                rec_score.append(self.recommendations[rec_based_on][i][0])
+                rec_item_id.append(self.recommendations[rec_based_on][i][1])
+                rec_title.append(self.recommendations[rec_based_on][i][2])
+
+        
+        # Save the output
+        output_dict = {self.id_col: item_id,
+                    'title': title,
+                    'rec_rank': rec_rank,
+                    'rec_score': rec_score,
+                    'rec_'+self.id_col: rec_item_id,
+                    'rec_title': rec_title}
+
+        # Convert to dataframe
+        self.top_k_recommendations = pd.DataFrame(output_dict)
+    
+    def recommend_top_k_items(self, df_clean):
```

Comment:
minor comment, in other files we add here the `k` instead of in the init. Normally it is not a part of the variables of the class

Reply:
Yes, that makes much more sense. This change has been made.